dask/dataframe/reshape.py

```diff
@@ -161,19 +163,33 @@ def get_dummies(
 
     # We explicitly create `meta` on `data._meta` (the empty version) to
     # work around https://github.com/pandas-dev/pandas/issues/21993
-    meta = pd.get_dummies(
-        data._meta,
-        prefix=prefix,
-        prefix_sep=prefix_sep,
-        dummy_na=dummy_na,
-        columns=columns,
-        sparse=sparse,
-        drop_first=drop_first,
-        **kwargs
-    )
+    if isinstance(data._meta, (pd.Series, pd.Index, pd.DataFrame)):
+        meta = pd.get_dummies(
+            data._meta,
+            prefix=prefix,
+            prefix_sep=prefix_sep,
+            dummy_na=dummy_na,
+            columns=columns,
+            sparse=sparse,
+            drop_first=drop_first,
+            **kwargs
+        )
+        method_cache = pd
+    else:
+        meta = M.get_dummies(
+            data._meta,
+            prefix=prefix,
+            prefix_sep=prefix_sep,
+            dummy_na=dummy_na,
+            columns=columns,
+            sparse=sparse,
+            drop_first=drop_first,
+            **kwargs
+        )
+        method_cache = M
```

Comment:
A few questions/comments here:

1.  It seems odd to special case Pandas here.  Ideally we would find a single solution that would work for any pandas-like library
2.  I don't understand the use of `M` here.  The `M` object creates methods, so for example this would be equivalend to `data._meta.get_dummies`.  My understanding is that this doesn't exist for Pandas.

@TomAugspurger , do you have any suggestions for how to get a function or a module of an object to go from a Pandas dataframe to a Pandas function (not method).

Reply:
True, because of point 2 I had to have a branched condition for pandas separately. But I would like to know if we can do it in a more common code like you asked @TomAugspurger. I'm aswell looking to see how I can find package level function as both pandas & cudf now have `get_dummies` as a package level functions.

Reply:
> Pandas dataframe to a Pandas function (not method).
 
Sorry, I don't quite follow. Is it fair to say that in e.g. `dask.dataframe.get_dummies`, we'd like to get at `DataFrame.get_dummies`? Does `getattr(data, <name>)` work?

Reply:
Unfortuantely not in this case

```python
In [1]: import pandas as pd

In [2]: df = pd.DataFrame()

In [3]: df.get_dummies
AttributeError: 'DataFrame' object has no attribute 'get_dummies'

In [4]: pd.get_dummies
Out[4]: <function pandas.core.reshape.reshape.get_dummies(data, prefix=None, prefix_sep='_', dummy_na=False, columns=None, sparse=False, drop_first=False, dtype=None)>
```

Reply:
@TomAugspurger, like Matthew mentioned I'm trying to access package level function since `get_dummies` is not a DataFrame function in pandas. 
```python
>>> import cudf
>>> cudf.get_dummies
<function get_dummies at 0x7f0b55f63ae8>

```

Reply:
Ah understood.

Is this too hacky?

```
In [34]: obj = pd.DataFrame()

In [35]: package = obj.__class__.__module__.split(".")[0]

In [36]: sys.modules[package].get_dummies
Out[36]: <function pandas.core.reshape.reshape.get_dummies(data, prefix=None, prefix_sep='_', dummy_na=False, columns=None, sparse=False, drop_first=False, dtype=None)>
```

Reply:
It's certainly hacky :)  But also pragmatic :/

I think I'll leave this question to you if you don't mind :)

Reply:
I can live with it. I don't think it's likely to break (famous last words).

Reply:
Made changes accordingly.